### DISCUSSION TITLE
Nesting config files in virtual folders

### DISCUSSION THREAD

**Discussion by hinell** [nvim-tree/nvim-tree.lua#1963] [Ideas]
[![](https://user-images.githubusercontent.com/11247099/157142238-b00deecb-8d56-424f-9b20-ef6a6f5ddf99.png)](https://github.com/antfu/vscode-file-nesting-config)
The above is a VSCode plugin ([file nesting config](https://github.com/antfu/vscode-file-nesting-config)). The config hells is a well known issue of tools keeping their dot files at the root of the project. Would be handy to gather them into virtual folders.

**Comment by alex-courtis** [comment]
That would indeed be very useful functionality. Unfortunately it is _very_ specific and does not belong in nvim-tree, which is a file explorer, not an IDE.

How could we possibly achieve such functionality in nvim-tree? Add API to nvim-tree to allow such behaviour. That would not be simple and it is unlikely ever to be built.

Practical solution: use a separate, for-purpose plugin such as [tagbar](https://github.com/preservim/tagbar) (tags) or [vista.vim](https://github.com/liuchengxu/vista.vim) (LSP) that deals with individual files in the desired manner.

**Comment by hinell** [comment]
@alex-courtis I'm fine with that! You are doing great job anyway!
